File: programmers/level1/python/영어끝말잇기.py
def solution(n, words):
    used = []
    lastchar = words[0][0]

    who, turn = 0, 0

    for i, word in enumerate(words):
        if word[0] != lastchar or word in used:
            who = i % n + 1
            turn = int(i / n) + 1
            break

        used.append(word)
        lastchar = word[-1]

    return who, turn


print(solution(2, ["hello", "one", "even", "never", "now", "world", "draw"]))

File: programmers/level1/java/영어끝말잇기.java
package java;

import java.util.HashSet;

class Solution {
    public int[] solution(int n, String[] words) {
        HashSet<String> used = new HashSet<>();
        char lastchar = words[0].charAt(0);
        int who = 0;
        int turn = 0;

        for (int i = 0; i < words.length; i++) {
            String word = words[i];
            if (word.charAt(0) != lastchar || used.contains(word)) {
                who = i % n + 1;
                turn = (i / n) + 1;
                break;
            }

            used.add(word);
            lastchar = word.charAt(word.length() - 1);
        }

        int[] answer = { who, turn };
        return answer;
    }
}